[Failing Test] post-kubernetes-push-image-etcd

### Which jobs are failing?

master-informing:
- post-kubernetes-push-image-etcd

### Which tests are failing?

Overall job

### Since when has it been failing?

First failure: https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/post-kubernetes-push-image-etcd/1718020233660207104

### Testgrid link

https://testgrid.k8s.io/sig-release-master-informing#post-kubernetes-push-image-etcd

### Reason for failure (if possible)

```
ERROR: (gcloud.builds.submit) build 0bbd94cb-0c8b-41f0-9631-53590a87a4c7 completed with status "FAILURE"
```

Some additional suspicions:
```
/bin/bash: line 1: ./build.sh: No such file or directory
done
install: can't stat '/workspace/tmp.BbIfCf/etcd': No such file or directory
install: can't stat '/workspace/tmp.BbIfCf/etcdctl': No such file or directory
install: can't stat '/workspace/tmp.ilEfpe/etcd': No such file or directory
install: can't stat '/workspace/tmp.ilEfpe/etcdctl': No such file or directory

cp: cannot stat 'bin/arm64/etcd*': No such file or directory
....
```

### Relevant SIG(s)

/sig k8s-infra

https://kubernetes.slack.com/archives/C2C40FMNF/p1698666130888699 discussion in slack.

> looks like the error is [gcr.io/k8s-staging-etcd/etcd:3.5.10-0-linux-amd64](http://gcr.io/k8s-staging-etcd/etcd:3.5.10-0-linux-amd64) is a manifest list

> [gcr.io/k8s-staging-etcd/etcd:3.5.10-0-linux-amd64](http://gcr.io/k8s-staging-etcd/etcd:3.5.10-0-linux-amd64) is a 
> manifest list
> make: *** [Makefile:223: push-manifest] Error 1
> ERROR
> ERROR: build step 0 “[gcr.io/k8s-staging-test-infra/gcb-docker-gcloud:v20230623-56e06d7c18](http://gcr.io/k8s-staging-test-infra/gcb-docker-gcloud:v20230623-56e06d7c18)” failed: step exited with non-zero status: 2

--- a/cluster/images/etcd/Makefile
+++ b/cluster/images/etcd/Makefile
@@ -192,6 +192,7 @@ endif
 	# And build the image
 	docker buildx build \
 		--pull \
+		--provenance=false \
 		--output=type=$(OUTPUT_TYPE) \
 		--platform "$(OS)/$(ARCH)" \
 		-t $(REGISTRY)/etcd:$(IMAGE_TAG)-$(IMAGE_SUFFIX) \

